src/main/java/com/google/api/codegen/config/GapicMethodConfig.java

```diff
@@ -243,6 +243,40 @@ static GapicMethodConfig createMethodConfig(
     }
   }
 
+  @VisibleForTesting
+  static ResourceNameTreatment defaultResourceNameTreatment(
+      MethodConfigProto methodConfigProto,
+      Method method,
+      ProtoParser protoParser,
+      String defaultPackageName) {
+
+    ResourceNameTreatment defaultResourceNameTreatment =
+        methodConfigProto.getResourceNameTreatment();
+    if (defaultResourceNameTreatment == ResourceNameTreatment.UNSET_TREATMENT
+        && method
+            .getInputMessage()
+            .getFields()
+            .stream()
+            .anyMatch(
+                f ->
+                    !Strings.isNullOrEmpty(protoParser.getResourceReference(f))
+                        || !Strings.isNullOrEmpty(protoParser.getResourceOrSetEntityName(f)))) {
+      String methodInputPackageName =
+          protoParser.getProtoPackage(((ProtoFile) method.getInputMessage().getParent()));
+      if (!defaultPackageName.equals(methodInputPackageName)) {
```

Comment:
I think it's cleaner to flip the logic so equals comes first.

Reply:
for sure